File: dotnet-desktop-guide/winforms/advanced/snippets/how-to-use-host-builder/csharp/GreetingService.cs
using Microsoft.Extensions.Configuration;

namespace HostBuilderApp;

// <GreetingService>
public class GreetingService : IGreetingService
{
    private readonly IConfiguration _configuration;

    public GreetingService(IConfiguration configuration)
    {
        _configuration = configuration;
    }

    public string GetGreeting()
    {
        return _configuration.GetValue<string>("GreetingMessage")
            ?? "Hello, World!";
    }
}
// </GreetingService>

File: dotnet-desktop-guide/winforms/advanced/snippets/how-to-use-host-builder/vb/GreetingService.vb
Imports Microsoft.Extensions.Configuration

' <GreetingService>
Public Class GreetingService
    Implements IGreetingService

    Private ReadOnly _configuration As IConfiguration

    Public Sub New(configuration As IConfiguration)
        _configuration = configuration
    End Sub

    Public Function GetGreeting() As String Implements IGreetingService.GetGreeting
        Dim message As String = _configuration.GetValue(Of String)("GreetingMessage")

        If message Is Nothing Then
            Return "Hello, World!"
        End If

        Return message
    End Function

End Class
' </GreetingService>
